src/java/org/stathissideris/ascii2image/text/TextGrid.java

```diff
@@ -1,1786 +1,2094 @@
+  }
+
+  public CellSet getPointMarkersOnLine() {
+    CellSet result = new CellSet();
+    int width = getWidth();
+    int height = getHeight();
+    for (int yi = 0; yi < height; yi++) {
+      for (int xi = 0; xi < width; xi++) {
+        if (!isInPlainMode(xi, yi))
+          continue;
+        char c = get(xi, yi);
+        if (StringUtils.isOneOf(c, pointMarkers)
+            && isStarOnLine(new Cell(xi, yi))) {
+          result.add(new Cell(xi, yi));
+        }
+      }
+    }
+    return result;
+  }
+
+
+  public void _replaceHumanColorCodes() {
+    int height = getHeight();
+    for (int y = 0; y < height; y++) {
+      String row = rows.get(y).toString();
+      Iterator it = humanColorCodes.keySet().iterator();
+      while (it.hasNext()) {
+        String humanCode = (String) it.next();
+        String hexCode = (String) humanColorCodes.get(humanCode);
+        if (hexCode != null) {
+          humanCode = "c" + humanCode;
+          hexCode = "c" + hexCode;
+          row = row.replaceAll(humanCode, hexCode);
+          rows.set(y, new StringBuilder(row)); //TODO: this is not the most efficient way to do this
+          row = rows.get(y).toString();
+        }
+      }
+    }
+  }
+
+  public void replaceHumanColorCodes() {
+    int height = getHeight();
+    for (int y = 0; y < height; y++)
+      rows.set(y, replaceHumanColorCodes(y, rows.get(y)));
+  }
+
+  private StringBuilder replaceHumanColorCodes(int rowIndex, StringBuilder in) {
+    Pattern p = Pattern.compile("(c.{0,3}|[^c]+)");
+    StringBuilder ret = new StringBuilder(in.length());
+    Iterator<String> i = createTextSplitter(p, in);
+    while (i.hasNext()) {
+      String next = i.next();
+      if (isInPlainMode(ret.length(), rowIndex) && looksColorCode(next)) {
+        String replacement = humanColorCodes.get(next.substring(1, 4));
+        if (replacement != null)
+          ret.append(String.format("c%s", replacement));
+        else
+          ret.append(next);
+      } else
+        ret.append(next);
+    }
+    return ret;
+  }
+
+  private static boolean looksColorCode(String word) {
+    return word.length() >= 4 && word.startsWith("c");
+  }
+
+  /**
+   * Replace all occurrences of c1 with c2
+   *
+   * @param c1
+   * @param c2
+   */
+  public void replaceAll(char c1, char c2) {
+    int width = getWidth();
+    int height = getHeight();
+    for (int yi = 0; yi < height; yi++) {
+      for (int xi = 0; xi < width; xi++) {
+        char c = get(xi, yi);
+        if (c == c1)
+          set(xi, yi, c2);
+      }
+    }
+  }
+
+  public boolean hasBlankCells() {
+    CellSet set = new CellSet();
+    int width = getWidth();
+    int height = getHeight();
+    for (int y = 0; y < height; y++) {
+      for (int x = 0; x < width; x++) {
+        Cell cell = new Cell(x, y);
+        if (isBlank(cell))
+          return true;
+      }
+    }
+    return false;
+  }
+
+
+  public CellSet getAllNonBlank() {
+    CellSet set = new CellSet();
+    int w
```

Comment:
Fixing whitespace is great, but put such changes in a different pull request as they obscure the substance of your contribution and make the diff harder to read

Reply:
Got it, too. I'm fixing it soon. For the time being, please resort to "Hide white space changes", which is found under "Diff settings" button on a pull-request screen.